include/grpc/grpc.h

```diff
@@ -355,6 +355,20 @@ typedef struct grpc_op {
   } data;
 } grpc_op;
 
+
+/** Registers a plugin to be initialized and deinitialized with the library.
```

Comment:
1) change "deinitialized" to "destroyed". Same for the argument "deinit" -> "destroy"
2) Replace all but the intro line with the following:

> The \a init and \a destroy functions will be invoked as part of \a grpc_init() and \a grpc_shutdown(), respectively. Note that these functions can be invoked an arbitrary number of times (and hence so will \a init and \a destroy). 
> It is safe to pass NULL to either argument. Plugins are destroyed in the reverse order they were initialized.

The last line is only true if you implement Nico's comment to use an array instead of the current linked list to hold the registered plugins.